Bio/PDB/PDBList.py

```diff
@@ -701,88 +702,134 @@ def get_seqres_file(self, savefile="pdb_seqres.txt"):
+    )
+
+    # Add mutually exclusive arguments for mode
+    mode = parser.add_mutually_exclusive_group(required=True)
+    mode.add_argument(
+        "-update",
+        action="store_true",
+        help="write weekly PDB updates to local pdb tree",
+    )
+    mode.add_argument(
+        "-all", action="store_true", help="write all PDB entries to local pdb tree"
+    )
+    mode.add_argument(
+        "-obsol",
+        action="store_true",
+        help="write all obsolete PDB entries to local pdb tree",
+    )
+    mode.add_argument(
+        "-assemb",
+        action="store_true",
+        help="write all assemblies for each PDB entry to local pdb tree",
+    )
+    mode.add_argument(
+        "-codes",
+        metavar="",
+        help="retrieve structure(s) - accepts a single PDB code or a comma separated list of PDB codes",
+    )
+
+    # Add optional format argument that select choice
+    parser.add_argument(
+        "-format",
+        choices=("pdb", "xml", "mmtf"),
+        default="mmCif",
+        help="format in which to download structures, defaults to mmCif",
+    )
+
+    # Add final options
+    others = parser.add_argument_group()
+    others.add_argument(
+        "-d",
+        action="store_true",
+        help="a single directory will be used as pdb_path, not a tree",
+    )
+    others.add_argument(
+        "-o", action="store_true", help="overwrite existing structure files"
+    )
+    others.add_argument(
+        "-with-assemblies",
+        action="store_true",
+        help="downloads assemblies along with regular entries",
+    )
+
+    # Parse arguments
+    args = parser.parse_args()
+
+    # Set path
+    pdb_path = args.pdb_path
+    file_format = args.format
+
+    # Instantiate PDBList class
+    pl = PDBList(pdb=pdb_path)
+
+    # Set options
+    if (args.d) or (args.pdb_path == os.getcwd()):
+        pl.flat_tree = True
+    overwrite = args.o
+    with_assemblies = args.with_assemblies
+
+    # Run
+    if args.update:
+        # Update pdb
+        print("updating local PDB at " + pdb_path)
+        pl.update_pdb(file_format=file_format, with_assemblies=with_assemblies)
+
+    elif args.all:
+        # Get the entire PDB
+        pl.download_entire_pdb(file_format=file_format)
+        if with_assemblies:
+            # Get all assembly structures
             pl.download_all_assemblies(file_format=file_format)
 
-        elif len(sys.argv[1]) == 4 and sys.argv[1][0].isdigit():
-            pdb_code = sys.argv[1]
-            # get single PDB entry
+    elif args.obsol:
+        # Get all obsolete entries
+        pl.download_obsolete_entries(pdb_path, file_format=file_format)
+
+    elif args.assemb:
+        # Get all assembly structures
+        pl.download_all_assemblies(file_format=file_format)
+
+    elif args.codes:
+        if "," in args.codes:
+            pdb_ids = args.codes.split(",")
+        else:
+            pdb_ids = [args.codes]
```

Comment:
```suggestion
        pdb_ids = args.codes.split(",")
```

will do the same :v:

Reply:
So weird, I must have written something funny that was iterating over the string before. Introduced the check because it was splitting my single PDB code into individual characters. Whoops haha